main.hpp

```diff
@@ -51,162 +56,190 @@
 #include <boost/filesystem/operations.hpp>
 #endif
 
-
 namespace po = boost::program_options;
 using namespace Reservoir::WellIndexCalculation;
 using namespace std;
 
-void printCsv(vector<IntersectedCell> &well_blocks) {
-  cout << "i,\tj,\tk,\twi" << endl;
-  for (auto block : well_blocks) {
-    auto line = boost::str(boost::format("%d,\t%d,\t%d,\t%s")
-                               % (block.ijk_index().i() + 1)        // %1
-                               % (block.ijk_index().j() + 1)        // %2
-                               % (block.ijk_index().k() + 1)        // %3
-                               % block.well_index());               // %4
-    cout << line << endl;
-  }
+const double minimum_well_index = 1e-9;
+
+void printCsv(map<string, vector<IntersectedCell>> &well_indices) 
+{
+    cout << "i,\tj,\tk1,\tk2,\twi" << endl;
+
+    vector<string> well_names;
+    for(map<string, vector<IntersectedCell>>::iterator it = well_indices.begin(); it != well_indices.end(); ++it)
+    {
+      well_names.push_back(it->first);
+    }
+
+    for (auto well_name : well_names)
+    {
+		cout << well_name << endl;
+		for (auto block : well_indices[well_name])
+		{
+			if (block.cell_well_index() > minimum_well_index)
+			{
+				auto line = boost::str(boost::format("%d,\t%d,\t%d,\t%d,\t%s")
+						%(block.ijk_index().i() + 1)         // %1
+						%(block.ijk_index().j() + 1)         // %2
+						%(block.ijk_index().k() + 1)         // %3
+						%(block.ijk_index().k() + 1)         // %3
+						%block.cell_well_index());           // %4
+				cout << line << endl;
+			}
+		}
+    }
 }
 
-void printCompdat(vector<IntersectedCell>& well_blocks,
-                  string well_name,
-                  double wellbore_radius) {
-  string head = "COMPDAT\n";
-  string foot = "\n/";
-  vector<string> body;
-  //                      NAME   I   J  K1  K2  OP/SH ST  WI   RADIUS
```

Comment:
I think we should re-add this link, one extra line but it enables the following code to be more compact